src/core/Object3D.js

```diff
@@ -835,5 +849,79 @@ Object3D.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 } );
 
+function Vector3WithDirtyFlag( x, y, z ) {
```

Comment:
Is there any other good places where to define this new object, for example `src/math/Vector3WithDirtyFlag.js`?

Reply:
I mentioned this in the other PR but there's the option of encapsulating this logic entirely in the constructor of `Object3D` so we don't have another vector type. The use of the this type might be pretty specific to the `Object3D` class.

Reply:
No performance overhead?

Reply:
That's a good point -- I'm sure it does incur a bit more overhead on instantiation. I'm not sure how much, though.